### DISCUSSION TITLE
Unknown 'check_constraints' table

### DISCUSSION THREAD

**Discussion by arthurcornil** [symfony/symfony#52752] [Q&A] [answered]
## Hello!👋
I hope this message finds you well. I have been using Symfony for some time now, and have run into an issue with Doctrine that I haven't encountered before

### Problem summary
When attempting to run `symfony console make:migration`, I encounter the following error:
```
Base table or view not found: 1109 Unknown table 'check_constraints' in information_schema
```
### Context
- Symfony version: 6.1
- Doctrine version: 2.17
- MySQL version: 5.7

### Request for assistance
If anyone has encountered a similar issue or has insights into the possible causes, I would be grateful for your guidance. Additionally, if there are specific steps or configurations I should review to resolve this problem, your suggestions would be highly appreciated.

Thank you in advance for your time and assistance. If more information is needed, please let me know, and I'll be happy to provide it.

**Comment by MatTheCat** [comment]
Hello

please run commands using the `-vvv` flag so more informations are available.
Also, note that “Doctrine” encompasses many libraries, eg. you probably refer to `doctrine/orm` when you say your version is 2.17 (which don’t tell if you are using 2.17.0 or 2.17.1).

That being said, the only occurrence of `information_schema.CHECK_CONSTRAINTS` is found in `doctrine/dbal`’s `MariaDb1043Platform`. As you’re using MySQL 5.7 that probably means you have configured a wrong `serverVersion` for your connection (see https://www.doctrine-project.org/projects/doctrine-dbal/en/3.7/reference/configuration.html#automatic-platform-version-detection).

If it’s not the case, please reply my answer with the output of `composer show doctrine/dbal`.

**Reply by arthurcornil** [reply]
Thank you for your prompt response and valuable suggestions.

I have followed your advice and executed the Symfony commands with the `-vvv` flag to enable verbose mode. While doing so, I ensured to capture more detailed information.

Regarding the Doctrine version, you're correct, and I appreciate the clarification.

As for the MySQL server version, what bugs me is that I use the same `serverVersion` in the `.env` file of other projects. These other projects utilise identical versions of MariaDB and `doctrine/orm` and encounter no issues when running `php bin/console make:migration`.

Additionally, here is the output of `comsposer show doctrine/dbal`:
```
name     : doctrine/dbal
descrip. : Powerful PHP database abstraction layer (DBAL) with many features for database schema introspection and management.
keywords : abstraction, database, db2, dbal, mariadb, mssql, mysql, oci8, oracle, pdo, pgsql, postgresql, queryobject, sasql, sql, sqlite, sqlserver, sqlsrv
versions : * 3.7.2
type     : library
license  : MIT License (MIT) (OSI approved) https://spdx.org/licenses/MIT.html#licenseText
homepage : https://www.doctrine-project.org/projects/dbal.html
source   : [git] https://github.com/doctrine/dbal.git 0ac3c270590e54910715e9a1a044cc368df282b2
dist     : [zip] https://api.github.com/repos/doctrine/dbal/zipball/0ac3c270590e54910715e9a1a044cc368df282b2 0ac3c270590e54910715e9a1a044cc368df282b2
path     : /Users/arthur/Documents/Web/wecodx_ia/wecodx_ia/vendor/doctrine/dbal
names    : doctrine/dbal

support
issues : https://github.com/doctrine/dbal/issues
source : https://github.com/doctrine/dbal/tree/3.7.2

autoload
psr-4
Doctrine\DBAL\ => src

requires
composer-runtime-api ^2
doctrine/cache ^1.11|^2.0
doctrine/deprecations ^0.5.3|^1
doctrine/event-manager ^1|^2
php ^7.4 || ^8.0
psr/cache ^1|^2|^3
psr/log ^1|^2|^3

requires (dev)
doctrine/coding-standard 12.0.0
fig/log-test ^1
jetbrains/phpstorm-stubs 2023.1
phpstan/phpstan 1.10.42
phpstan/phpstan-strict-rules ^1.5
phpunit/phpunit 9.6.13
psalm/plugin-phpunit 0.18.4
slevomat/coding-standard 8.13.1
squizlabs/php_codesniffer 3.7.2
symfony/cache ^5.4|^6.0
symfony/console ^4.4|^5.4|^6.0
vimeo/psalm 4.30.0
```
Thank you for your continued assistance.

**Reply by stof** [reply, marked as answer]
Well, your initial message says you are using MySQL 5.7, not MariaDB. So it looks weird that the MariaDB platform of DBAL would be used.

Make sure you don't configure a MariaDB server version if you don't actually use MariaDB. Even though MariaDB implements the driver protocol of MySQL, the platform features are not the same.

**Reply by MatTheCat** [reply]
> I use the same `serverVersion` in the `.env` file of other projects. These other projects utilise identical versions of MariaDB

MySQL and MariaDB are two different softwares. If your project uses MySQL 5.7 then its `serverVersion` must reflect it.

**Reply by arthurcornil** [reply]
@stof, you're absolutely right. My initial message indeed mentioned MySQL 5.7, and it's clear now that configuring MariaDB in `serverVersion` was causing the issue. Thanks a lot for enlightening me. What I still struggle to understand is that many other projects I have, run the exact same versions and have a `serverVersion` mentioning MariaDb. There still is something that I don't quite understand.

**Reply by arthurcornil** [reply]
@MatTheCat, thanks a lot for your insights
